packages/core/admin/admin/src/pages/AuthPage/index.js

```diff
@@ -157,6 +159,16 @@ const AuthPage = ({ hasAdmin, setHasAdmin }) => {
 
       auth.setToken(token, false);
       auth.setUserInfo(user, false);
+      const { roles } = user;
+
+      if (roles) {
+        const isUserSuperAdmin = roles.find(({ code }) => code === 'strapi-super-admin');
+
+        if (isUserSuperAdmin) {
+          persistStateToLocaleStorage.setGuidedTourVisible(true);
+          setGuidedTourVisibility(true);
+        }
+      }
```

Comment:
Adding this code causes `setSubmitting(false)` on line 189 to fire a `Can't perform a React state update on an unmounted component` log error.

Questioning this setSubmitting Formik function:
Documentation https://formik.org/docs/api/formik says
`IMPORTANT: If onSubmit is async, then Formik will automatically set isSubmitting to false on your behalf once it has resolved. This means you do NOT need to call formikBag.setSubmitting(false) manually. However, if your onSubmit function is synchronous, then you need to call setSubmitting(false) on your own.`

Since our onSubmit `registerRequest` is async, is setSubmitting really needed?
If it is, could we fix it by modifying the way we call onSubmit in Register page by adding async/await there?
```
    onSubmit={async (data, formik) => {
      if (registrationToken) {
        // We need to pass the registration token in the url param to the api in order to submit another admin user
        await onSubmit({ userInfo: omit(data, ['registrationToken']), registrationToken }, formik);
      } else {
        await onSubmit(data, formik);
      }
    }}
```